panels/site_sections/panel_applications.py

```diff
@@ -32,34 +55,72 @@ def index(self, session, message='', **params):
-                elif other_panelists and 'verify_poc' not in params:
-                    message = 'You must agree to being the point of contact for your group'
-                else:
-                    session.add_all([app, panelist] + other_panelists)
-                    raise HTTPRedirect('index?message={}', 'Your panel application has been submitted')
+                message = process_panel_app(session, app, panelist, **params)
+                if not message:
+                    raise HTTPRedirect('guest?poc_id={}message={}', poc_id, 'Your panel application has been submitted')
 
         return {
             'app': app,
             'message': message,
             'panelist': panelist,
-            'ops_count': len(other_panelists),
-            'other_panelists': other_panelists,
+            'other_panelists': other_panelists_from_params(app, **params),
             'verify_tos': params.get('verify_tos'),
-            'verify_poc': params.get('verify_pos'),
+            'verify_poc': params.get('verify_poc'),
             'verify_waiting': params.get('verify_waiting'),
             'verify_unavailable': params.get('verify_unavailable')
         }
+
+    def guest(self, session, poc_id, message='', **params):
+        """
+        In some cases, we want pre-existing attendees (e.g., guests) to submit panel ideas.
+        This submission form bypasses the need to enter in one's personal and contact info
+        in favor of having the panel application automatically associated with an attendee
+        record, both as the submitter and as the Point of Contact.
+        """
+
+        app = session.panel_application(params, checkgroups=PanelApplication.all_checkgroups, restricted=True, ignore_csrf=True)
+        app.poc_id = poc_id
+        attendee = session.attendee(id=poc_id)
+        panelist = PanelApplicant(
+            app_id=app.id,
+            attendee_id=attendee.id,
+            submitter=True,
+            first_name=attendee.first_name,
+            last_name=attendee.last_name,
+            email=attendee.email,
+            cellphone=attendee.cellphone
+        )
+
+        if cherrypy.request.method == 'POST':
+            message = process_panel_app(session, app, panelist, **params)
+            if not message:
+                raise HTTPRedirect('guest?poc_id={}&message={}', poc_id, 'Your panel application has been submitted')
+
+        return {
+            'app': app,
+            'message': message,
+            'attendee': attendee,
+            'poc_id': poc_id,
+            'other_panelists': other_panelists_from_params(app, **params),
+            'verify_unavailable': params.get('verify_unavailable')
+        }
+
+
+def process_panel_app(session, app, panelist, **params):
+    """
+    Checks various parts of a new panel application, either submitted by guests or by non-attendees,
+    and then adds them to a session.
+    """
+    other_panelists = other_panelists_from_params(app, **params)
```

Comment:
This all was a bit odd. I chose to do this as a way to get around a problem I had passing `other_panelists` as an argument to the `process_panel_app` function. The function would try to unpack the list, resulting in a "too many arguments" error. So rather than doing that, and rather than calling the `other_panelists_from_params` function several times, we only grab the other panelists when we actually need them -- when rendering the forms (in the return statements above) and when processing the app here.

Reply:
Whoops, I mentioned this above, but don't you end up creating them twice?

Reply:
Yup... I couldn't figure out a way around the error that I mentioned -- in fact, that same one you got. :\ This was a way to work around that error since I was really stuck on it, but clearly the fact that you ran into it means I need to actually figure it out.